ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2

```diff
@@ -8,6 +8,35 @@ router isis {{ router_isis.instance }}
 {%     if router_isis.is_type is arista.avd.defined %}
    is-type {{ router_isis.is_type }}
 {%     endif %}
+{%     for redistribute_route in router_isis.redistribute_routes | arista.avd.natural_sort('source_protocol') %}
+{%         if redistribute_route.source_protocol is arista.avd.defined %}
+{%             set redistribute_route_cli = "redistribute " ~ redistribute_route.source_protocol %}
+{%             if redistribute_route.source_protocol == 'isis' %}
+{%                 set redistribute_route_cli = redistribute_route_cli ~ " instance" %}
+{%             elif redistribute_route.source_protocol == 'ospf' %}
+{%                 if redistribute_route.include_leaked is arista.avd.defined(true) %}
+{%                     set redistribute_route_cli = redistribute_route_cli ~ " include leaked" %}
+{%                 endif %}
+{%                 if redistribute_route.ospf_route_type is not arista.avd.defined %}
+{%                     continue %}
+{%                 endif %}
+{%                 set redistribute_route_cli = redistribute_route_cli ~ " match " ~ redistribute_route.ospf_route_type %}
+{%             elif redistribute_route.source_protocol == 'ospfv3' %}
+{%                 if redistribute_route.ospf_route_type is not arista.avd.defined %}
+{%                     continue %}
+{%                 endif %}
+{%                 set redistribute_route_cli = redistribute_route_cli ~ " match " ~ redistribute_route.ospf_route_type %}
```

Comment:
```suggestion
{%                 if redistribute_route.ospf_route_type is arista.avd.defined %}
{%                     set redistribute_route_cli = redistribute_route_cli ~ " match " ~ redistribute_route.ospf_route_type %}
{%                 endif %}
```

Reply:
Since ospf_route_type is mandatory for redistribution there is no need to render anymore if it isn't defined. With ```continue```, an all or nothing behavior seems cleaner.

Reply:
Reverted logic to use continue